xcodeproj/internal/xcodeproj_rule.bzl

```diff
@@ -617,14 +644,29 @@ targets.
             for id in target_merge_srcs_by_label[src_target.label]:
                 target_merges.pop(id, None)
 
+    # Remap 'target_merge_dests' after popping invalid merges.
+    target_merge_dests = {}
     for src, dests in target_merges.items():
-        src_target = focused_targets.pop(src)
-
         for dest in dests:
-            focused_targets[dest] = xcode_targets.merge(
-                src = src_target,
-                dest = focused_targets[dest],
-            )
+            target_merge_dests.setdefault(dest, []).append(src)
+
+    for dest, srcs in target_merge_dests.items():
+        src_targets = [focused_targets.pop(src) for src in srcs]
+
+        src_target_swift = None
+        src_target_objc = None
+
+        for src_target in src_targets:
+            if src_target.is_swift:
```

Comment:
FYI, with https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2105, you'll need to change this to look at `src_target._swift_params` (probably renaming it to not have an underscore).